include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -917,7 +1050,7 @@ namespace VectorTools
       additional_data.mapping_update_flags = (update_values | update_JxW_values);
       MatrixFree<dim, Number> matrix_free;
       matrix_free.reinit (mapping, dof, constraints,
-                          QGauss<1>(fe_degree+1), additional_data);
+                          QGauss<1>(fe_degree+2), additional_data);
```

Comment:
please, don't change this here. For projection from quadrature data, the number of quadrature points is up to the user. Even though the mass matrix may not be integrated exactly, in most cases you would use `fe_degree+1`. So I am not in favour of this change and below.